### ISSUE TITLE
Argument to force CuDNN implementation for tf.keras.layers.LSTM

### ISSUE THREAD

**Issue by jkamalu** [tensorflow/tensorflow#30535]
**System information**
- TensorFlow version (you are using): 2 beta1
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

Boolean `force_cudnn` argument for tf.keras.layers.LSTM. When True, would raise an error if the user's configurations or usage (masking) are incompatible with the CuDNN implementation. 

Currently, the LSTM layer logic dynamically uses the CuDNN implementation depending on the given arguments to the `__init__` and `call` functions. It does not report which it ends up using.

**Will this change the current api? How?**

This will add an argument to the tf.keras.layers.LSTM `__init__` function.

**Who will benefit with this feature?**

Anyone for whom the CuDNN implementation is both essential and viable and who is debugging their architecture for this reason. For example... "My model is very slow, but I haven't run it before. I wonder if the LSTM implementation is optimized for GPU? I wish I could verify that it is."

**Comment by rmothukuru** [comment]
@jkamalu,
Sorry for the delayed response. As per the documentation of [tf.keras.layers.LSTM](https://www.tensorflow.org/api_docs/python/tf/keras/layers/LSTM), it uses **`GPU`** if the below conditions are satisfied.

> activation == tanh
> recurrent_activation == sigmoid
> recurrent_dropout == 0
> unroll is False
> use_bias is True
> Inputs, if use masking, are strictly right-padded.
> Eager execution is enabled in the outermost context.

Can you please let us know if this is what you are looking for? Thanks!
